tools/src/main/java/org/apache/kafka/tools/TopicCommand.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools;
+
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.MutuallyExclusiveGroup;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.ListTopicsOptions;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.admin.TopicListing;
+import org.apache.kafka.common.errors.TopicExistsException;
+import org.apache.kafka.common.internals.Topic;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+import static net.sourceforge.argparse4j.impl.Arguments.storeTrue;
+
+/**
+ * Command class for operations on topics
+ */
+public class TopicCommand {
+
+    private AdminClient client;
+
+    public static void main(String[] args) throws Exception {
+
+        new TopicCommand().run(args);
+    }
+
+    void run(String[] args) throws Exception {
+
+        TopicCommandOptions opts = new TopicCommandOptions(args);
+
+        if (args.length == 0)
+            CommandLineUtils.printUsageAndDie(opts.parser, "Create, delete, describe, or change a topic.");
+
+        opts.checkArgs();
+
+        Properties props = new Properties();
+        props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, opts.brokerList());
+
+        this.client = AdminClient.create(props);
```

Comment:
Good point ! That's true but at same time we have to be sure that the client creation and initialization happes only when the parsing and checking arguments goes well without error/exception. It could mean not having the Admin client initialized in the constructor of the base abstract class but having a method that the derived class should call after the arguments checks.